rtl/eclrtl/CMakeLists.txt

```diff
@@ -79,16 +79,8 @@ include_directories (
 
 ADD_DEFINITIONS( -D_USRDLL -DECLRTL_EXPORTS )
 
-if (USE_BOOST_REGEX)
-    if ((CMAKE_COMPILER_IS_GNUCXX OR CMAKE_COMPILER_IS_CLANGXX) AND (${BOOST_REGEX_VERSION} LESS 105700) AND 
-        (ICU_VERSION VERSION_LESS "10.0"))
-        set_source_files_properties(eclregex.cpp PROPERTIES COMPILE_FLAGS -std=c++98)
-    endif ()
-
-    if (WIN32)
-    else ()
-        ADD_DEFINITIONS( -DBOOST_DYN_LINK )
-    endif ()
+if ((USE_BOOST_REGEX) AND (NOT WIN32))
+  ADD_DEFINITIONS( -DBOOST_DYN_LINK )
```

Comment:
@richardkchapman 
I think this would mean that we do not support ICU versions < 60 - although earlier versions of the library sources may have been patched to fix the problem.
What is the oldest version that we support (centos6? centos7?) and does that compile with this fix?
It looks like centos 7 may default to v50, which would appear to be likely to hit the issue.

Reply:
7 is the oldest CentOS version we're currently supporting and yes, it only has libicu v50 in the repos.  We don't have issues on centOS8 with v60.  If I remove the c++98 standard for that single file, it will compile fine on centos7.